File: windows/src/ConnectorControl.Core/State/ImportChoice.cs
namespace ConnectorControl.Core.State;

/// <summary>
/// What one connector of an imported document does about the connector the target collection
/// already has under that name. <see cref="Add"/> is the case where nothing is in the way. Each
/// caller picks its own default for a collision: Import leaves the row unticked, which sends
/// <see cref="Skip"/>, and preselects Replace should it be ticked; Copy preselects Keep both,
/// AppState's own default.
///
/// Mirror: Sources/ConnectorControlState/ImportChoice.swift
/// </summary>
public enum ImportChoice
{
    Add,
    Replace,
    KeepBoth,
    Skip,
}

File: Sources/ConnectorControlState/ImportChoice.swift
/// What one connector of an imported document does about the connector the target collection
/// already has under that name. `add` is the case where nothing is in the way. Each caller picks
/// its own default for a collision: Import leaves the row unticked, which sends `skip`, and
/// preselects Replace should it be ticked; Copy preselects Keep both, AppState's own default.
///
/// Mirror: windows/src/ConnectorControl.Core/State/ImportChoice.cs
public enum ImportChoice: String, Equatable, Sendable {
    case add, replace, keepBoth, skip
}
